packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -1271,3 +1280,183 @@ export function validateEventListenerTarget(
   }
   return false;
 }
+
+export const supportsTestSelectors = true;
+
+export function findRootFiber(node: Instance): null | Fiber {
+  const stack = [node];
+  while (stack.length > 0) {
+    const current = stack.shift();
```

Comment:
Nit: Increment an index (O(1)) instead of `shift`-ing (O(n) in the worst case)

Reply:
Sure :smile: This PR isn't really ready for review btw.

Reply:
Haha missed that! Ok I'll stop :D